dbms/src/Flash/Coprocessor/ExecutionSummary.h

```diff
@@ -25,6 +26,9 @@ struct ExecutionSummary
     UInt64 num_produced_rows = 0;
     UInt64 num_iterations = 0;
     UInt64 concurrency = 0;
+
+    std::shared_ptr<DB::DM::TableScanContext> table_scan_context = std::make_shared<DB::DM::TableScanContext>();
```

Comment:
why not use `DB::DM::TableScanContext table_scan_context` directly?
Or `std::optional` and `std::unique_ptr`

Reply:
I think unique_ptr may be a better choice.